workflows/dags/data/data_harvest.py

```diff
@@ -0,0 +1,221 @@
+        )
+
+        return hepdata_response.json()
+
+    @task_group
+    def process_record(record_id):
+        """
+        Process the record by downloading the versions,
+        building the record and loading it to inspirehep.
+        """
+
+        @task(task_concurrency=5)
+        def download_record_versions(id):
+            """
+            Download the versions of the record.
+
+            Args: id (int): The id of the record.
+            Returns: dict: The record versions.
+            """
+
+            hepdata_response = generic_http_hook.call_api(
+                endpoint=f"/record/ins{id}?format=json"
+            )
+            hepdata_response.raise_for_status()
+            payload = hepdata_response.json()
+
+            record = {"base": payload}
+            for version in range(1, payload["record"]["version"]):
+                response = generic_http_hook.call_api(
+                    endpoint=f"/record/ins{id}?format=json&version={version}"
+                )
+                record[version] = response.json()
+
+            return record
+
+        @task.virtualenv(
+            requirements=["inspire-schemas"],
+            system_site_packages=False,
+            map_index_template="{{params}}",
+        )
+        def build_record(data_schema, payload, **context):
+            """Build the record from the payload.
+
+            Args: data_schema (str): The schema of the data.
+                    payload (dict): The payload of the record.
+
+            Returns: dict: The built record.
+            """
+            import datetime
+            import re
+
+            from inspire_schemas.builders import DataBuilder
+
+            def add_version_specific_dois(record, builder):
+                """
+                Add dois to the record.
+                """
+                for data_table in record["data_tables"]:
+                    builder.add_doi(data_table["doi"], material="part")
+                for resource_with_doi in record["resources_with_doi"]:
+                    builder.add_doi(resource_with_doi["doi"], material="part")
+
+                builder.add_doi(record["record"]["hepdata_doi"], material="version")
+
+            def add_keywords(record, builder):
+                """
+                Add keywords to the record.
+                """
+                for keyword, item in record.get("data_keywords", {}).items():
+                    if keyword == "cmenergies":
+                        if len(item) >= 1 and "lte" in item[0] and "gte" in item[0]:
+                            builder.add_keyword(
+                                f"{keyword}: {item[0]['lte']}-{item[0]['gte']}"
+                            )
+                    elif keyword == "observables":
+                        builder.add_keyword(f"{keyword}: {','.join(item)}")
+                    else:
+                        for value in item:
+                            builder.add_keyword(value)
```

Comment:
can we move somewhere separate? for example utils?

Reply:
I will try, because its a virtual env I am not completely sure if its possible

Reply:
ok, I see. We can leave as is for now

Reply:
indeed I checked, and in the `sys.path` of this task only the following exist:
`['/tmp/venv-calliyoq65lm', '/usr/local/lib/python311.zip', '/usr/local/lib/python3.11', '/usr/local/lib/python3.11/lib-dynload', '/tmp/venvicyq6h_l/lib/python3.11/site-packages']`

its possible to do `sys.path.append('/opt/airflow/plugins'])` and it would work. although it would look jankier

Reply:
it's ok we can leave it like that, no need to append stuff to the path